### PR TITLE
feat(note-taking): added global-note.nvim plugin

### PR DESCRIPTION
**Pull request by Billuc**
## 📑 Description

Added a plugin for global-note.nvim

## ℹ Additional Information

Also added a command for opening a project-local note

### CODE REVIEW

File: lua/astrocommunity/note-taking/global-note-nvim/init.lua [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  local project_name = vim.fs.basename(project_directory)
+  if project_name == nil then
+    vim.notify("Unable to get the project name", vim.log.levels.WARN)
+    return nil
+  end
+
+  return project_name
+end
```
Comment (mehalter) [inline lua/astrocommunity/note-taking/global-note-nvim/init.lua:15]:
```suggestion
local get_project_name = function()
  local project_name

  local project_directory, err = vim.loop.cwd()
  if not project_directory then
    vim.notify(err, vim.log.levels.WARN)
  else
    project_name = vim.fs.basename(project_directory)
  end

  if not project_name then
    vim.notify("Unable to get the project name", vim.log.levels.WARN)
  end

  return project_name
end
```

Recommendation to simplify logic of function for calculating the project name

**Review by mehalter — COMMENTED** [review]
This looks great! And I love the inclusion of the project local note as well. Added a comment with some small nits for the function for calculating the project name. Let me know what you think or if there's a problem with it.

File: lua/astrocommunity/note-taking/global-note-nvim/init.lua (hunk 1 above)
Comment (Billuc) [reply lua/astrocommunity/note-taking/global-note-nvim/init.lua]: Yup, no problem with that ! FYI, the previous code was given by the author of the global-note.nvim repo :)

File: lua/astrocommunity/note-taking/global-note-nvim/init.lua [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  local project_directory, err = vim.loop.cwd()
+  if not project_directory then
+    vim.notify(err, vim.log.levels.WARN)
+  else
+    project_name = vim.fs.basename(project_directory)
+  end
+
+  if not project_name then
+    vim.notify("Unable to get the project name", vim.log.levels.WARN)
+  end
```
Comment (mehalter) [inline lua/astrocommunity/note-taking/global-note-nvim/init.lua:13]:
```suggestion
  if not project_name then vim.notify("Unable to get the project name", vim.log.levels.WARN) end
```

This should fix the formatting with `stylua`

**Review by mehalter — CHANGES_REQUESTED** [review]

**Review by mehalter — APPROVED** [review]

File: lua/astrocommunity/note-taking/global-note-nvim/init.lua (hunk 2 above)
Comment (Billuc) [reply lua/astrocommunity/note-taking/global-note-nvim/init.lua]: Yeah I have seen it, I have modified the commit

**Pull request merged**